R/CohortDefinition.R

```diff
@@ -516,19 +516,46 @@ invokeCohortSetGeneration <- function(baseUrl, sourceKeys, definitionIds) {
 getCohortInclusionRulesAndCounts <- function(baseUrl, 
                                              cohortId, 
                                              sourceKey) {
-  url <- sprintf("%s/cohortdefinition/%d/report/%s?mode=0",
-                 baseUrl, cohortId, sourceKey)
-  json <- httr::GET(url)
-  json <- httr::content(json) 
+  urlByEvent <- sprintf("%s/cohortdefinition/%d/report/%s?mode=0",
+                        baseUrl, cohortId, sourceKey)
+  jsonByEvent <- httr::GET(urlByEvent)
+  jsonByEvent <- httr::content(jsonByEvent) 
+  jsonByEventTreeMapData <- jsonByEvent$treemapData
+  
+  
+  urlByPerson <- sprintf("%s/cohortdefinition/%d/report/%s?mode=1",
+                         baseUrl, cohortId, sourceKey)
+  jsonByPerson <- httr::GET(urlByPerson)
+  jsonByPerson <- httr::content(jsonByPerson) 
+  jsonByPersonTreeMapData <- RJSONIO::fromJSON(jsonByPerson$treemapData)
+  
+  summaryByEventDf <- as.data.frame(jsonByEvent$summary)
+  summaryByPersonDf <- as.data.frame(jsonByPerson$summary)
   
-  results <- lapply(json$inclusionRuleStats, function(j) {
-    list(ruleId = j$id,
-         description = j$name,
-         indexPersonCount = json$summary$baseCount,
-         rulePersonCount = j$countSatisfying,
-         rulePercentSatisfied = j$percentSatisfying,
-         rulePercentToGain = j$percentExcluded,
-         matchRate = json$summary$percentMatched)  
+  inclusionRuleStatsByEventDf <- lapply(jsonByEvent$inclusionRuleStats, function(j) {
+    list(id = j$id,
+         name = j$name,
+         countSatisfying = j$countSatisfying,
+         percentSatisfying = j$percentSatisfying,
+         percentExcluded = j$percentExcluded)  
   })
-  do.call(rbind.data.frame, results)
+  inclusionRuleStatsByEventDf <- do.call(rbind.data.frame, inclusionRuleStatsByEventDf)
+  
+  inclusionRuleStatsByPersonDf <- lapply(jsonByPerson$inclusionRuleStats, function(j) {
+    list(id = j$id,
+         name = j$name,
+         countSatisfying = j$countSatisfying,
+         percentSatisfying = j$percentSatisfying,
+         percentExcluded = j$percentExcluded)  
+  })
+  inclusionRuleStatsByPersonDf <- do.call(rbind.data.frame, inclusionRuleStatsByPersonDf)
+  
+  resultsByEvent <- list(summary = summaryByEventDf, 
+                         inclusionRuleStats = inclusionRuleStatsByEventDf
+  )
+  resultsByPerson <- list(summary = summaryByPersonDf, 
+                          inclusionRuleStats = inclusionRuleStatsByPersonDf
+  )
+  results <- list(resultsByEvent = resultsByEvent, resultsByPerson = resultsByPerson
```

Comment:
To follow convention, remove the "results <-" part

Reply:
```suggestion
  list(resultsByEvent = resultsByEvent, resultsByPerson = resultsByPerson
```